File: Golang/ex12.go
package main

import (
	"fmt"
)

type Produto struct {
	Nome  string
	Preco float64
}

func SomarProdutos(p1, p2 Produto) Produto {
	return Produto{
		Nome:  p1.Nome + " e " + p2.Nome,
		Preco: p1.Preco + p2.Preco,
	}
}

func main() {
	produto1 := Produto{Nome: "Produto A", Preco: 30.50}
	produto2 := Produto{Nome: "Produto B", Preco: 20.75}

	resultado := SomarProdutos(produto1, produto2)

	fmt.Printf("Produto: %s\nPreço Total: %.2f\n", resultado.Nome, resultado.Preco)
}

File: python/ex12.py
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    def __add__(self, other):
        if isinstance(other, Produto):
            
            return Produto(f"{self.nome} & {other.nome}", self.preco + other.preco)
        return NotImplemented

    def __str__(self):
        return f"{self.nome}: R${self.preco:.2f}"


def main():
    produto1 = Produto("Produto A", 50.00)
    produto2 = Produto("Produto B", 30.00)

    
    produto_soma = produto1 + produto2

    
    print(produto1)
    print(produto2)
    print(f"Soma: {produto_soma}")


if __name__ == "__main__":
    main()
